### DISCUSSION TITLE
feat: Add Thinking Time Indicator for Improved User Feedback in o1-Series API Responses

### DISCUSSION THREAD

**Discussion by Adhithya03** [open-webui/open-webui#5932] [Ideas]
I would like to suggest adding a **thinking time indicator** that displays the time in seconds it takes for the o1-series (or other similar reasoning models in the future) to generate a response from the API, similar to the feature available in OpenAI's official ChatGPT.
 
- Although this is a minor enhancement, it would significantly improve user experience by providing visual feedback that the model is actively working on the request. Currently, with streaming responses disabled, users might perceive the API request as indefinitely paused. Displaying the processing time would reassure users that progress is being made.

**Additional context**

![image](https://github.com/user-attachments/assets/79cd65d1-6c2c-48d8-a390-fd70cfc174c4)

**Comment by yusuf-madkour** [comment]
It's not just the visual feedback, is it possible to actually get the "thought" process the model went through? In OAI interface, when I click on the arrow beside `Thought for X seconds`, I could see how the model got the answer but AFAIK, this is not possible in open-webui.

**Reply by Ryan526** [reply]
You could probably "fake" the thought process using a cheaper model but that's just a waste of money/compute.

I'm pretty sure the thought process shown when using the OAI interface is also mocked. They don't want you to see its actual thought process.

**Comment by Ryan526** [comment]
@Adhithya03 
> I would like to suggest adding a **thinking time indicator** that displays the time in seconds it takes for the o1-series (or other similar reasoning models in the future) to generate a response from the API, similar to the feature available in OpenAI's official ChatGPT.


I wanted the same thing so I built a function for this. Don't enable it globally, just enable it for models that don't support streaming like o1. Maybe in the future this can be built in as the default way to handle models with streaming disabled.

https://openwebui.com/f/rjalynski/thinkingui
https://github.com/Ryan526/Thinking

## Example Output

- During processing:
![image](https://github.com/user-attachments/assets/e13774ae-9336-4184-8181-65c608cb93aa)

- After the response is processed:
![image](https://github.com/user-attachments/assets/78bc2be0-a04a-4c65-951d-a6a4caf2da02)

**Reply by Adhithya03** [reply]
This is super cool! Thanks very much.
